api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -4003,3 +4003,166 @@ func TestValidateAPIServerAccessProfile(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateAMCPVirtualNetwork(t *testing.T) {
+	tests := []struct {
+		name      string
+		amcp      *AzureManagedControlPlane
+		expectErr bool
```

Comment:
Can we make this more specific than a bool so we can better verify the kind of error being returned here?

Reply:
added via 6b1927121631e5ca59ef6dd42e4005af4ad506e1